cmake/modules/AddSwift.cmake

```diff
@@ -246,40 +245,39 @@ function(_add_host_variant_c_compile_flags target)
       # the build.
       if(CMAKE_C_COMPILER_ID MATCHES Clang AND CMAKE_C_COMPILER_VERSION
           VERSION_LESS 9.0.0)
-        target_compile_options(${target} PRIVATE -mcx16)
+        target_compile_options(${target} PRIVATE $<$<COMPILE_LANGUAGE:C,CXX,OBJC,OBJCXX>:-mcx16>)
       endif()
     endif()
   endif()
 
   if(LLVM_ENABLE_ASSERTIONS)
-    target_compile_options(${target} PRIVATE -UNDEBUG)
+    target_compile_options(${target} PRIVATE $<$<COMPILE_LANGUAGE:C,CXX,OBJC,OBJCXX>:-UNDEBUG>)
   else()
-    target_compile_definitions(${target} PRIVATE -DNDEBUG)
+    target_compile_options(${target} PRIVATE $<$<COMPILE_LANGUAGE:C,CXX,OBJC,OBJCXX>:-DNDEBUG>)
```

Comment:
I by mistake changed this to definitions. I am going to change this back before it goes in. (I think I did it in a few more places as well).

Reply:
I looked through the commit and these were the only two that I found.

Reply:
This technically changes the behaviour - it changes ordering to be precise.  I think that using `target_compile_definitions` is better, but Im not opposed to this change.

Reply:
Was just an error